includes/rest-api/class-sensei-rest-api-import-controller.php

```diff
@@ -42,4 +42,166 @@ protected function get_handler_class() {
+				],
+				[
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => [ $this, 'request_delete_file' ],
+					'permission_callback' => [ $this, 'can_user_access_rest_api' ],
+				],
+				'schema' => [ $this, 'get_file_item_schema' ],
+			]
+		);
+	}
+
+	/**
+	 * Handle the request to upload a file.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function request_post_file( $request ) {
+		$files = $request->get_file_params();
+		if ( ! isset( $files['file'] ) ) {
+			return new WP_Error(
+				'sensei_data_port_invalid_file',
+				__( 'No file was uploaded', 'sensei-lms' ),
+				[ 'status' => 400 ]
+			);
+		}
+
+		$data_port_manager = Sensei_Data_Port_Manager::instance();
+		$job               = $data_port_manager->get_active_job( $this->get_handler_class(), get_current_user_id() );
+		if ( ! $job ) {
+			$job = $this->create_job();
+		}
+
+		// Check if the job has started or if there was an error creating the job.
+		if ( ! $job || $job->is_started() ) {
+			return new WP_Error(
+				'sensei_data_port_job_started',
+				__( 'Job has already been started', 'sensei-lms' ),
+				[ 'status' => 400 ]
+			);
+		}
+
+		// Check to make sure the upload succeeded.
+		if (
+			! isset( $files['file']['tmp_name'], $files['file']['error'] )
+			|| ! $this->is_uploaded_file( $files['file']['tmp_name'] )
+			|| UPLOAD_ERR_OK !== $files['file']['error']
+		) {
+			return new WP_Error(
+				'sensei_data_port_job_upload_failed',
+				__( 'Upload was not successful', 'sensei-lms' )
+			);
+		}
+
+		$result = $job->save_file( $request->get_param( 'file_key' ), $files['file']['tmp_name'], $files['file']['name'] );
+
+		if ( is_wp_error( $result ) ) {
+			return $result;
+		}
+
+		$response = new WP_REST_Response();
+		$response->set_status( 200 );
+		$response->set_data( $this->prepare_to_serve_job( $job ) );
+
+		return $response;
+	}
+
+	/**
+	 * Check if a file was uploaded.
+	 *
+	 * @param string $filename Temporary file path to check.
+	 *
+	 * @return bool
+	 */
+	private function is_uploaded_file( $filename ) {
+		// Disable this check in tests as it isn't possible to bypass. This is the constant
+		// WordPress core uses to see if we're within tests.
+		if ( defined( 'DIR_TESTDATA' ) && DIR_TESTDATA ) {
+			return true;
+		}
+
+		return is_uploaded_file( $filename );
+	}
+
+	/**
+	 * Handle the request to delete a file.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function request_delete_file( $request ) {
+		$data_port_manager = Sensei_Data_Port_Manager::instance();
+		$job               = $data_port_manager->get_active_job( $this->get_handler_class(), get_current_user_id() );
+		if ( ! $job ) {
+			return new WP_Error(
+				'sensei_data_port_job_does_not_exist',
+				__( 'No active job has been found', 'sensei-lms' ),
```

Comment:
Since these are displayed to the user, can we add full stops?

Reply:
9cb3b94af3c20dce506905d6f331f32c7be381ae